### DISCUSSION TITLE
Output as float

### DISCUSSION THREAD

**Discussion by dryginmo** [davidfrantz/force#179] [Q&A] [answered]
I´m aiming to use TSA NDVI data to train a 2D CNN in Tensor Flow. I´m faced with the problem that AI training data requires value ranges between 0 and 1, thus making it necessary to convert the NDVI to float and rescale it. Since I also need to use equivalent scaled data for my model predictions on a whole Sentinel 2 scene it probably would speed up my processing if the TSA module allows a floating output. Is there any possibility to make this happen?

Thanks for your feedback,
Johannes

**Comment by davidfrantz** [comment, marked as answer]
Hi Johannes,

the short answer is: no, this is not possible with FORCE. 

Setting the output to float would be OK to do, I guess. But the thing that complicates this beyond a point that is reasonable, is that user-defined scaling functions would need to be provided, along with options that define a range limit, as well as options that control the behavior when scaling extends beyond these limits, as well as user-defined nodata values...

Different parameters would be needed for all the different indices and different output products... These could be pre-defined, but I doubt that there is a choice that fits all use cases. Implementing user-defined parameters for all of these would blow up the parameter file, which is already quite packed...

All in all, this is simply too much to be reasonably implemented into the core routine, and a custom conversion script might still be the better choice.
Such a script could be developed and shipped with FORCE, though. Please feel free to contribute such a scrip if you like ;)

Cheers,
David
